cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+        int index = 0;
+        StringBuffer partitionList = new StringBuffer("");
+        List<String> sourcePartitionColumnList = new ArrayList<>();
+        while (rsPartitionList.next()) {
+            if (rsPartitionList.getString(1).equals("# Partition Information")) {
+                index++;
+            }
+            if (index > 0) index++;
+            if (index > 4) {
+                String partitionNameAndDataType = rsPartitionList.getString(1) + " " + rsPartitionList.getString(2).toUpperCase();
+                partitionList.append(partitionNameAndDataType + ",");
+                sourcePartitionColumnList.add(partitionNameAndDataType);
+            }
+        }
+
+        sourceColumnList.removeAll(sourcePartitionColumnList);
+        //removing partition columns from the total columns list and concatenating with commas
+        StringBuffer finalColumns = new StringBuffer("");
+        for (String sourceColumn : sourceColumnList) {
+            finalColumns.append(sourceColumn + ",");
+        }
+        sourceRegularColumns=finalColumns.substring(0, finalColumns.length() - 1);
+        sourcePartitionColumns=partitionList.substring(0, partitionList.length() - 1);
+        stgPartitionColumns=sourcePartitionColumns.substring(0,sourcePartitionColumns.lastIndexOf(","))+","+bdreTechPartition;
+        stgTableDDL = "create external table " + sourceDb + "." + sourceStgtable + " (" + sourceRegularColumns + ") " + "partitioned by (" + stgPartitionColumns + ") stored as orc location '/tmp/"+processId+"/"+instanceExecId+"'";
+        destTableDDL = "create table " + destDb + "." + table  + " (" + sourceRegularColumns + ") " + "partitioned by (" + stgPartitionColumns + ") stored as orc";
+        LOGGER.debug("stgTableDDL = " + stgTableDDL);
+        LOGGER.debug("destTableDDL = " + destTableDDL);
+        execStageTableDDL(st,sourceDb,sourceStgtable);
+        rsPartitionList.close();
+    }
+
+    private void execStageTableDDL(Statement st,String sourceDb, String sourceStgtable) throws Exception{
+        st.executeUpdate("drop table if exists " + sourceDb + "." + sourceStgtable);
+        st.executeUpdate(stgTableDDL);
+    }
+
+    private boolean checkIfDestTableExists(String destDb,String table, String destHiveConnection) throws Exception{
+        Connection conn = getHiveJDBCConnection(destDb,destHiveConnection);
+        Statement st = conn.createStatement();
+        ResultSet rsPartitions = st.executeQuery("show tables");
+        boolean destTableExists=false;
+        while (rsPartitions.next()) {
+            if(rsPartitions.getString(1).equalsIgnoreCase(table)) {
+                destTableExists=true;
+                break;
+            }
+        }
+        rsPartitions.close();
+        st.close();
+        conn.close();
+        return destTableExists;
+    }
+
+    private void alterDestTable(Statement st, List<String> addedColumnList, String destDb,String table) throws Exception{
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00112)